services/invitations/src/simcore_service_invitations/core/settings.py

```diff
@@ -0,0 +1,78 @@
+from functools import cached_property
+from typing import Optional, cast
+
+from pydantic import Field, HttpUrl, PositiveInt, SecretStr, validator
+from settings_library.base import BaseCustomSettings
+from settings_library.basic_types import BuildTargetEnum, LogLevel, VersionTag
+from settings_library.utils_logging import MixinLoggingSettings
+
+from .._meta import API_VERSION, API_VTAG, PROJECT_NAME
+
+
+class BaseApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
+    # CODE STATICS ---------------------------------------------------------
+    API_VERSION: str = API_VERSION
+    APP_NAME: str = PROJECT_NAME
+    API_VTAG: VersionTag = API_VTAG
+
+    # IMAGE BUILDTIME ------------------------------------------------------
+    # @Makefile
+    SC_BUILD_DATE: Optional[str] = None
+    SC_BUILD_TARGET: Optional[BuildTargetEnum] = None
+    SC_VCS_REF: Optional[str] = None
+    SC_VCS_URL: Optional[str] = None
+
+    # @Dockerfile
+    SC_BOOT_TARGET: Optional[BuildTargetEnum] = None
+    SC_HEALTHCHECK_TIMEOUT: Optional[PositiveInt] = Field(
+        None,
+        description="If a single run of the check takes longer than timeout seconds "
+        "then the check is considered to have failed."
+        "It takes retries consecutive failures of the health check for the container to be considered unhealthy.",
+    )
+    SC_USER_ID: Optional[int] = None
+    SC_USER_NAME: Optional[str] = None
+
+    # RUNTIME  -----------------------------------------------------------
+
+    INVITATIONS_LOGLEVEL: LogLevel = Field(
+        LogLevel.INFO, env=["INVITATIONS_LOGLEVEL", "LOG_LEVEL", "LOGLEVEL"]
+    )
+
+    @cached_property
+    def LOG_LEVEL(self):
+        return self.INVITATIONS_LOGLEVEL
+
+    @validator("INVITATIONS_LOGLEVEL")
+    @classmethod
+    def valid_log_level(cls, value: str) -> str:
+        # NOTE: mypy is not happy without the cast
+        return cast(str, cls.validate_log_level(value))
+
+
+class DesktopApplicationSettings(BaseApplicationSettings):
```

Comment:
The idea is that you can start this program as a server or as an executable. In the first case it exposes an http API and runs forever. IN the second, you use the CLI to run, get the results and close.  I call the latter Desktop to differenciate it from Web.

Settings are separated because they have custom options. E.g. the web app has http basic auth and the desktop up doesn't.

Reply:
I refactored it.